### DISCUSSION TITLE
Glitchy Audio in recording

### DISCUSSION THREAD

**Discussion by jlpaquette** [ExistentialAudio/BlackHole#541] [Q&A]
I'm having an issue where recording audio will be fine for a little bit, then start sounding robotic and glitchy. I saw [this link in the WIki](https://github.com/ExistentialAudio/BlackHole#why-is-audio-glitching-after-x-minutes-when-using-a-multi-output-or-an-aggregate) that says Drift Correction needs to be enabled, which it is and I'm still having the issue. 

![Screen Shot 2022-03-04 at 6 53 12 AM](https://user-images.githubusercontent.com/12281589/156785765-d27ce1b5-a248-40de-8a37-8386f36a95d3.png)

Note I've tried with both the built-in MacBook Pro Speakers, as well as my Thunderbolt Display speakers (Display Audio); both as the first devices, with Blackhole Drift Correction enabled. The audio sounds fine during recording, but replaying the recording is glitchy.

I had the same problem while using Soundflower before switching to BlackHole, so I'm wondering if it is some other audio settings or issues not necessarily BlackHole itself, but I don't really know where to look if that is the case. I've tried reboots and it'll be okay for a little while after a reboot and then the glitchiness comes back. It's not the content itself as it sounds fine during the recording process, and if it was glitchy, then I reboot, then I can record successfully... but glitchiness still comes back later on other recordings.

Any ideas what else I should be checking/changing to resolve?

**Comment by devinroth** [comment]
Unfortunately even with drift correction I’ve noticed some issues with aggregates.

Probably the best bet is to increase your audio buffer.

Devin
